pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +552,74 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+// Taking snapshot of cluster queue is enabled when maxcount non-zero
+func (r *ClusterQueueReconciler) isVisibilityEnabled() bool {
+	return r.queueVisibilityClusterQueuesMaxCount > 0
+}
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	if !r.isVisibilityEnabled() {
+		return nil
+	}
+	if r.shouldUpdatePendingWorkloadStatus(cq) {
+		return &kueue.ClusterQueuePendingWorkloadsStatus{
+			Head:           r.qManager.GetSnapshot(cq.Name),
```

Comment:
Here is actually a small performance issue we may fix in follow up, as this code may compute the snapshot twice: 
- once to check the condition, and once to put the head. 
I guess, we could do only the initial checking by `cq.Status.PendingWorkloadsStatus == nil || time.Since(cq.Status.PendingWorkloadsStatus.LastChangeTime.Time) >= r.queueVisibilityUpdateInterval`. If this check pass we need to compute the snapshot and reuse as value.

Reply:
I returned the previous iteration to avoid the same problem with performance. If we want to make this condition more beautiful, it makes sense to do it in another PR

Reply:
sgtm, you may just want to address https://github.com/kubernetes-sigs/kueue/pull/1069/commits/d50c9e18c0a1b6109e789f5db3f1b396194629d6#r1318830209 in this PR.

Reply:
Oh, I seem the Head is already returned. Feel free to mark as resolved.